### DISCUSSION TITLE
Understanding SentenceRecognizer

### DISCUSSION THREAD

**Discussion by julkhami** [explosion/spaCy#9847] [Help: Coding & Implementations] [answered]
I have some questions about this documentation: https://spacy.io/api/sentencerecognizer

What is a “pipeline component”? Is everything in the nlp() object considered part of some “pipeline”?

What are “assigned attributes”? As opposed to just an attribute? In what way it is “assigned”?

Does it suffice to just do:

doc = nlp(str)
sentences = list(doc.sents)

?

Thanks very much

**Comment by adrianeboyd** [comment, marked as answer]
Hi, I'd suggested starting with this overview in the docs: https://spacy.io/usage/spacy-101, in particular this section on pipelines: https://spacy.io/usage/spacy-101#pipelines
